File: baekjoon/2606.py
n = int(input())
m = int(input())
t = n + 1
adj = [[0 for _ in range(t)] for _ in range(t)]
for _ in range(m):
    x, y = map(int, input().split())
    adj[x][y] = 1
    adj[y][x] = 1
visited = [False for _ in range(t)]

cnt = 0

def dfs(v):
    global cnt
    visited[v] = True
    cnt += 1
    for i in range(1, t):
        if not visited[i] and adj[i][v] == 1:
            dfs(i)


#dfs(1)
#print(cnt - 1)


def bfs(v):
    cnt = 0
    queue = [v]
    visited[v] = True
    while queue:
        v = queue.pop(0)
        for i in range(1, t):
            if not visited[i] and adj[i][v] == 1:
                queue.append(i)
                visited[i] = True
                cnt += 1
    return cnt
    
print(bfs(1))

File: baekjoon/cpp/2606.cpp
#include <iostream>
#define MAX 101
#include <queue>
using namespace std;
int map[101][101] = {0,};
bool visited[101] = {0,};
int main(){
    int n, v;
    cin >> n;
    cin >> v;
    for (int i = 0; i < v; i++){
        int a, b;
        cin >> a >> b;
        map[a][b] = 1;
        map[b][a] = 1;
    }
    visited[1] = true;
    queue<int> q;
    q.push(1);
    int cnt = 0;
    while (!q.empty()){
        v = q.front();
        q.pop();
        for (int i = 1; i <= n; i++){
            if (map[v][i] == 1 && visited[i] == 0){
                q.push(i);
                visited[i] = true;
                cnt++;
            }
        }
    }
    cout << cnt << endl;
}